src/LaunchDarkly.EventSource/EventSourceService.cs

```diff
@@ -107,13 +107,22 @@ public EventSourceService(Configuration configuration)
                                 }
 
                                 processResponse(readLineTask.Result);
+
+                                if (closeOnEndOfStream && reader.EndOfStream)
+                                {
+                                    break;
+                                }
                             }
                         }
                     }
 
                     OnConnectionClosed();
                 }
             }
+            catch (HttpRequestException e) {
+                OnConnectionClosed();
```

Comment:
You shouldn't need to call the OnConnectionClosed method here. The calling code handles basic Exception types and will call Connection Close - see the catch block in EventSource.cs in the ConnectToEventSourceAsync method: `CloseAndRaiseError(e);`

Reply:
I believe `OnConnectionClosed` should be the method passed into EventSource @ initialization time. EventSource.cs does close the client on error, but does not call `OnConnectionClosed`

Reply:
EventSource.cs doesn't need to call OnConnectionClosed, rather the EventSourceService.cs calls this on the EventSource.cs. In the event of an error, EventSource itself will set the Ready State to Closed.